src/odemis/gui/util/img.py

```diff
@@ -112,12 +108,60 @@ def format_rgba_darray(im_darray, alpha=None):
         rgba[:, :, 1] = im_darray[:, :, 1]
         rgba[:, :, 2] = im_darray[:, :, 0]
         rgba[:, :, 3] = im_darray[:, :, 3]
-        new_darray = model.DataArray(rgba)
-        new_darray.metadata['byteswapped'] = True
-        return new_darray
     else:
         raise ValueError("Unsupported colour depth!")
 
+    new_darray = model.DataArray(rgba)
+    new_darray.metadata['byteswapped'] = True
+    return new_darray
+
+
+def format_bgra_to_rgb(im_darray, keepalpha=True, inplace=False):
+    """ Reshape the given numpy.ndarray from RGB to BGRA format
```

Comment:
Is this function converting to rgb (as the name suggests) or to BGRA (as the first line of the docstring suggests)

Reply:
Indeed. I messed up the docstring. It's now fixed.